### PR TITLE
imp: adds more options to `dereplicate_sequences()`

### LINKED ISSUE #47: expose --minuniquesize for dereplicating
**Improvement Description**
A forum member [wanted to use --minuniquesize during dereplication](https://forum.qiime2.org/t/vsearch-dereplicate-seqs-error/2984/4?u=colinbrislawn). Can we expose this setting so users can omit low abundance clusters?

**Comments**
This issues applies directly to `qiime vsearch dereplicate-sequences`, but may apply to other vsearch functions as well.

**References**
[wanted to use --minuniquesize during dereplication](https://forum.qiime2.org/t/vsearch-dereplicate-seqs-error/2984/4?u=colinbrislawn)

### LINKED ISSUE #68: derep - expose minseqlength parameter
**Improvement Description**
Expose the `-minseqlength` vsearch parameter in the `dereplicate-sequences` method.

**References**
1. [forum](https://forum.qiime2.org/t/any-way-to-control-read-length-parameter-filtering-in-vserach-dereplicate/14611)

### PR DESCRIPTION
**Pull request by colinbrislawn**
EDIT:

Description: Add `--min_seq_length` and `--min_unique_size` settings to `vsearch dereplicate-sequences` to discard short or uncommon sequences from the dereplication output. 

---

Closes #47 `--minseqlength` for derep
Also closes #68 `--minuniquesize` for derep

Ready for review, all feedback welcome! 

EDIT: Hashing is hard, I'll do that in a second PR.

### CODE REVIEW

**Comment by lizgehret** [comment]
Thanks @colinbrislawn! We are doing some PR triage and review right now and will take a look at this soon.

File: q2_vsearch/plugin_setup.py [hunk 1]
```diff
@@ -249,6 +251,9 @@
                          'longer sequences, it is clustered with the shortest '
                          'of them. If they are equally long, it is clustered '
                          'with the most abundant.'),
+        'min_seq_length': ('Discard sequences shorer than this integer.'),
```
Comment (Oddant1) [inline q2_vsearch/plugin_setup.py]: Probably supposed to be shorter not shorer? Other than that this seems good. Just exposing more vsearch functionality right?

Comment (colinbrislawn) [reply q2_vsearch/plugin_setup.py]: Fixed!

Yes, this should be a simple PR, most of the commits are me messing with hashing or trying to get testing and linting to work.

**Comment by Oddant1** [comment]
@colinbrislawn, is this ready to merge?

**Comment by lizgehret** [comment]
Hey @colinbrislawn, one final request - do you mind adding in a quick description in this PR that describes what this does and who it affects so we can include that in the 2022.11 release changelog? 🙂

**Comment by colinbrislawn** [comment]
@lizgehret, done! Let me know how it looks!

**Comment by lizgehret** [comment]
That's perfect, thanks @colinbrislawn!

**Comment by colinbrislawn** [comment]
Looks like I messed up the line ending in `plugin_setup.py` in the other PR.

Whoops! I'll look into this and get it merged.

**Comment by colinbrislawn** [comment]
I think this is now ready to merge!

**Review by lizgehret — APPROVED** [review]
lgtm, thanks @colinbrislawn!

**Pull request merged**